File: Programmers/체육복.py
def solution(n, lost, reserve):
    students = [1] * n
    for i in range(len(lost)):
        students[lost[i] - 1] -= 1
    for i in range(len(reserve)):
        students[reserve[i] - 1] += 1

    if students[0] == 2:
        if students[1] == 0:
            students[0] -= 1
            students[1] += 1
    for i in range(1, n - 1):
        if students[i] == 2:
            if students[i - 1] == 0:
                students[i - 1] += 1
                students[i] -= 1
                continue
            if students[i + 1] == 0:
                students[i + 1] += 1
                students[i] -= 1
                continue
    if students[-1] == 2:
        if students[-2] == 0:
            students[-2] -= 1
            students[-1] += 1
    answer = n - students.count(0)
    return answer

File: Programmers JS/체육복.js
function solution(n, lost, reserve) {
  var answer;
  let students = new Array(n).fill(1);
  lost.forEach((el) => students[el - 1]--);
  reserve.forEach((el) => students[el - 1]++);

  for (let i = 0; i < students.length; i++) {
    if (students[i] === 0) {
      if (i === 0 && students[i + 1] === 2) {
        students[i]++;
        students[i + 1]--;
      } else if (i !== 0) {
        if (students[i - 1] > 1 || students[i + 1] > 1) {
          students[i]++;
          if (students[i - 1] === 2) {
            students[i - 1]--;
          } else {
            students[i + 1]--;
          }
        }
      }
    }
  }
  answer = students.filter((el) => el > 0);
  return answer.length;
}
